tests/unit/package_managers/test_pip.py

```diff
@@ -120,7 +118,7 @@ def test_get_pip_metadata(
     setup_cfg.get_name.return_value = cfg_name
     setup_cfg.get_version.return_value = cfg_version
 
-    mock_get_repo_id.return_value = MOCK_REPO_ID
+    mock_get_repo_name.return_value = "bar-subpath"
```

Comment:
I know this is not ideal, but I made minimum changes just to pass the CI.

The function is awful. Actually, I was the one who added pyproject.toml to the mix. A good first step in refactoring the pip module would be splitting the "metadata" (pyproject.toml, setup.py, setup.cfg) into a separate file.